src/arch/riscv/isa/templates/vector_mem.isa

```diff
@@ -644,14 +671,16 @@ Fault
     }
     %(op_decl)s;
     %(op_rd)s;
+    %(set_vlenb)s;
+    %(set_vlen)s;
     %(ea_code)s;
 
     Fault fault = readMemAtomicLE(xc, traceData, EA,
                                   *(vreg_t::Container*)(&Mem), memAccessFlags);
```

Comment:
Please specify the size of memory for read, the method will read the memory data with maximum vector length size.